NuRadioMC/utilities/muon_flux.py

```diff
@@ -0,0 +1,180 @@
+from NuRadioReco.utilities import units
+import numpy as np
+import os
+import pickle
+from scipy.interpolate import interp1d
+from MCEq.core import MCEqRun
+import crflux.models as crf
+from functools import lru_cache
+
+class MuonFlux:
+    def __init__(self):
+
+        self.mc_m = 1e2
+        self.mc_rad = 180. / np.pi
+        self.mc_eV = 1.e-9
+        self.mc_ns = 1e-9
+
+        self.__buffer = {}
+        self.file_buffer = "data/surface_muon_buffer.pkl"
+        if(os.path.exists(self.file_buffer)):
+            fin = open(self.file_buffer, "rb")
+            self.__buffer = pickle.load(fin)
+            fin.close()
+
+
+    @lru_cache(maxsize=5000)
+    def get_mu_flux(self, theta, altitude=3200, interaction_model='SIBYLL23C', primary_model=(crf.GlobalSplineFitBeta, None), particle_names=("total_mu+", "total_mu-")):
+        
+        """
+        The function get_mu_flux returns the muon flux at theta, for a given altitude, CR model, and hadronic interaction model. 
+                
+        Parameters
+        ----------
+        energy: float
+            energy in eV
+        theta_min: float
+            minimum zenith angle in rad
+        theta_max: float
+            maximum zenith angle in rad
+        n_steps: int
+            number of steps to use for the numerical integration
+
+        Returns
+        -------
+        e_grid: array of floats
+            energy grid in eV
+        flux: array of floats
+            flux in NuRadioReco units 1/(area * time  * energy * steradian)
+        """
+         
+        altitude *= self.mc_m
+        mceq = MCEqRun(interaction_model=interaction_model,
+                       primary_model=primary_model,
+                       theta_deg=theta * self.mc_rad)
```

Comment:
potentially dangerous, please also divide by `units.rad`. Or just directly divide by `units.deg` if you know that MCEq wants it in degrees.